provision.yml

```diff
@@ -7,147 +7,181 @@
     tags:
      - always
 
-
-- hosts: loadbalancer:loadbalancer-vm:php-apps:java-apps:storage:oidc:dbcluster:sysloghost:elk:lifecycle
+- hosts: base
   gather_facts: yes
   become: true
   roles:
-    - { role: common,       tags: ['common'] }
-    - { role: rsyslog,      tags: ['rsyslog'] }
-    - { role: iptables, when: iptables_enable | bool,  tags: ['iptables'] }
-
-- hosts: selfsigned_certs
-  gather_facts: no
-  become: true
-  roles:
-    - { role: selfsigned_certs, tags: ['selfsigned_certs'] } 
-
-- hosts: php-apps-vm
-  become: true
-  roles:
-    - { role: hosts,     tags: ['hosts'] }
-    - { role: httpd,     tags: ['httpd'] }
-    - { role: welcome,   tags: ['welcome'] }
-
-- hosts: loadbalancer-vm
-  gather_facts: true
-  become: true
-  roles:
-    - { role: haproxy, tags: ['lb'] }
+    - role: common
+      tags: ['core', 'base', 'common']
+    - role: rsyslog
+      tags: ['core', 'base', 'rsyslog']
+    - role: iptables
+      when:
+        - iptables_enable | bool
+      tags: ['core', 'base', 'iptables']
+    - role: selfsigned_certs
+      when:
+        - use_selfsigned_certs
+      tags: ['core', 'base', 'selfsigned_certs']
 
-- hosts: loadbalancer_ha
-  gather_facts: true
+- hosts: frontend
   become: true
   roles:
-    - { role: keepalived, tags: ['keepalived'] }
-    - { role: bind, tags: ['bind'] }
+    - role: hosts
+      tags: ['core', 'frontend', 'hosts']
```

Comment:
Maybe create a boolean here as well. The "hosts" role is only for first time deployers where no dns entries are present yet. If we can make this optional it's easier to disable it for hosts where it is not necessary (and could potentially break the installation)

Reply:
With first time deployers you don't mean deploying it for the first time and rerun the provision command? As the `hosts` role is idempotent so it won't add things twice.

Reply:
Is now to be deployed with a toggle.